### DISCUSSION TITLE
[PSQL] [ORM] Conditional Foreign Key Column?

### DISCUSSION THREAD

**Discussion by zikphil** [sqlalchemy/sqlalchemy#6134] [Usage Questions] [answered]
Hello Community!

Not sure if that's even possible with the SQLAlchemy ORM but here is my scenario:

I have `ModelA` which has a `type` field which can be either `foo` or `bar`.

In `ModelB` I have a `model_a_id` field which is a foreign key reference to `ModelA`.

What I am trying to do is to restrict `ModelB.model_a_id` so that only `ModelA.type == foo` ids can be in that field.

So in this scenarios, the `model_a_id` field of `ModelB` could only contain references to `ModelA` which are of type foo.

Any ideas on how to do this?

Thanks,

Phil

**Comment by zzzeek** [comment, marked as answer]
I was going to say, "triggers", which still can be what happens here and is likely the one technique that can handle UPDATEs as well; the triggers would refer to psql functions that check the parent table.     I searched a bit and also found this stackoverflow answer: https://stackoverflow.com/a/4898314    Now, the author of that answer is clearly pretty out there, but if you scroll way down he has links to example code for SQL server.    The basic idea is, use a CHECK constraint on the "modelb" table, that runs a custom PostgreSQL function to test that the parent row is legit.  but I dont see how that would work for UPDATEs of hte parent table, which points back to just using triggers.
